### DISCUSSION TITLE
Configuration/Decoding question

### DISCUSSION THREAD

**Discussion by gperdach** [john30/ebusd#663] [Q&A] [answered]
Hi,

I have a question, based on decoding messages with following message:
2022-10-01 07:24:01.886 [bus notice] <0015b524060200020002001a0006030202000100b500

Based on the config 15.700.csv, following lines will decode the message
*r,,,,,,B524,02000200,,,IGN:4,,,
r,,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,0200,,,mctype;IGN:1,,,
decoded result: 2022-10-01 07:24:01.881 [update notice] received read 700 Hc1CircuitType QQ=00: mixer

As i understood *r will add the columns to the next r-lines, if they are not replaced. So wenn i combine this, then i write
r,,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,0200,,,mctype;IGN:1,,,

First I got a missing circuit name error, then I tried
r,700,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,0200,,,mctype;IGN:1,,,
After this, I got no error, but above message will not be decoded.

Then I tried to combine ID - fields, because i don't know what happens with this in the code and tried
r,700,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,020002000200,,,mctype;IGN:1,,,
With this, message will also not be decoded.

Based on the WIKI (4.1 Message definition and eBUS background), the last message should be ok, because
00 15 b524 06 020002000200 1a0006030202000100b500 compared with ebusctl info should be
00..master
15..destination slave (file name 15.700.csv should be ok)
b524..primary and secondary command (column PBSB in 15.700.csv)
06..count of bytes following
After that, it is a little bit difficult, because 6 means count of bytes for ID, the message is hex, or? if the message is hex, there should be two bytes per character, so this should be, or am I wrong?
020..?? 
If this are bytes, then the definition should be 020000
But there is always 020002000200 in this message.
Why is the first definition working?

Can anyone help me to understand this?

Regards,
Gregor

**Comment by john30** [comment, marked as answer]
> *r,,,,,,B524,02000200,,,IGN:4,,,
> r,,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,0200,,,mctype;IGN:1,,,

this definition is using defaults from the first line (starting with "*r"), so it will take the file name into account and will be reduced to the following before being interpreted:
`r,700,Hc1CircuitType,CircuitType Heizkreis 1,,,B524,020002000200,,,IGN:4;mctype;IGN:1,,,`

in the message sequence, the remaining bytes are:
* 1a: crc
* 00: ack
* 06: answer length
* 030202000100: answer data
* b5: crc
* 00: ack

so the actual interesting data is `030202000100` and that is then interpreted like this:
* 03020200: skipped for "IGN:4"
* 01: mctype
* 00: skipped for "IGN:1"

**Comment by gperdach** [comment]
Thanks a lot.
The * notation is a little bit difficult to read.
`ebusctl find -f`
helped also to export the current configuration. With this, all * values will be replaced.
